### ISSUE TITLE
Error: Engine not found for the ".hbs" file extension

### ISSUE THREAD

**Issue by sean-hill** [niftylettuce/email-templates#276]
Hey there, I'm using `handlebars` as my templating engine. I have a `html.hbs` file in my project and have `handlebars` installed locally in my project. I have my options setup like this:

```js
const email = new Email({
  views: {
    root: path.resolve(__dirname, `templates`),
    options: {
      extension: 'hbs'
    }
  }
})
```

But I'm receiving a:

    Error: Engine not found for the ".hbs" file extension

error when trying to render the template. Any suggestions?

**Comment by sean-hill** [comment]
Using `.ejs` with the `ejs` templating engine is also returning an empty string as the rendered template.

**Comment by niftylettuce** [comment]
I think it might have to do with `handlebars` vs `hbs` extension.  I remember a similar issue with nunjucks engine in the past.  Unsure though, have you tried renaming your file extensions to `.handlebars`?

<https://github.com/tj/consolidate.js/blob/master/lib/consolidate.js#L171-L189>

**Comment by niftylettuce** [comment]
I believe I see exactly what issue is, we have to do a line similar to <https://github.com/tj/consolidate.js/blob/master/lib/consolidate.js#L1610>.

```js
exports.hbs = fromStringRenderer('handlebars');
```

**Comment by niftylettuce** [comment]
Want to test this out locally and submit a pull request to the consolidate repo?

Basically you'd do this:
1. Fork email-templates GitHub repo, clone it locally, go inside, run `yarn` and then `npm link`.
2. Fork consolidate GitHub repo, clone it locally, go inside, run `npm install`, and then `npm link`.
3. Go into email-templates folder, run `npm link consolidate`
4. Go into consolidate, add the code into the lib/consolidate.js file
5. Go into your project, run `npm link email-templates`, and then run it again
6. Test it out and if it works submit pull request to consolidate repo 

cc @sean-hill @polonel

**Comment by md-fileshadow** [comment]
Just to let you know I also have this issue. I've forked the repos and done the edit to consolidate, but still it does not work for me. I'm getting a "Cannot find module 'handlebars'" error, but I might not have managed my dependencies correctly. I will keep working on it but wanted you to know this really breaks my email system since all of my templates are using handlebars.

**Comment by niftylettuce** [comment]
`npm install handlebars` in the root of your project?

**Comment by md-fileshadow** [comment]
Yes, handlebars has been installed in the root of the project but still not working. But I found a workaround here:

https://github.com/niftylettuce/email-templates/issues/149

Renaming my template files with `.handlebars` extensions and adding the following to my code before doing a `renderAll()` works:

`// Override consolidate's handlebars instance
email.config.views.options.engineSource.requires.handlebars = Handlebars; `

Sorry I don't have more time to pursue this to get the .hbs extensions working, but thanks for the response!

**Comment by niftylettuce** [comment]
Should be resolved now, try v3.4.0.  I've added the default mapping to of hbs to handlebars per the PR in #284.

**Comment by niftylettuce** [comment]
@petersaints I've added this in v3.5.1 of email-templates.  The only thing you will need to do is specify the `views.options.extension: 'njk'` option.  The mapping of `njk: 'nunjucks'` has been added as a default similar to hbs => handlebars above.  `npm install email-templates@latest` or `yarn add email-templates@latest`.  thanks for request.
